meilisearch-types/src/document_formats.rs

```diff
@@ -36,13 +35,15 @@ impl fmt::Display for PayloadType {
 
 #[derive(Debug)]
 pub enum DocumentFormatError {
+    Io(io::Error),
     Internal(Box<dyn std::error::Error + Send + Sync + 'static>),
     MalformedPayload(Error, PayloadType),
 }
 
 impl Display for DocumentFormatError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
+            Self::Io(e) => write!(f, "{e}"),
```

Comment:
Do we want to say here that the error is usually (but not always) caused by "no space left on disk" or "permission errors"?

I'm also wondering if we could feasibly add more context here: on which file did the access operation fail, while carrying out which logical operation, etc. This imposes to store more than just the IoError though.

Reply:
Nice, good catch. I totally missed the `documentFormatError` thingy.

We don't need to because that's handled when we create the `ResponseError` from the `Code`